src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java

```diff
@@ -344,8 +346,8 @@ private int storeItemStack(@NotNull final ItemStack itemStackIn)
     {
         for (int i = 0; i < this.stacks.length; ++i)
         {
-            if (this.stacks[i] != null && this.stacks[i].getItem() == itemStackIn.getItem() && this.stacks[i].isStackable()
-                  && this.stacks[i].stackSize < this.stacks[i].getMaxStackSize() && this.stacks[i].stackSize < this.getInventoryStackLimit()
+            if (!ItemStackUtils.isEmpty(this.stacks[i]) && this.stacks[i].getItem() == itemStackIn.getItem() && this.stacks[i].isStackable()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (7) used in the expression (maximum allowed 6). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)